src/persist/src/indexed/mod.rs

```diff
@@ -596,11 +644,27 @@ impl<L: Log, B: Blob> Indexed<L, B> {
             }
         };
 
+        // Regardless of success or failure of writing meta, we need to remember that some
```

Comment:
changed this to remove from in_flight_compactions as soon as we get the response. I believe what changes now is that the compaction work gets more speculative.

if i have 3 batches [a, b, c,]

and i receive a compaction response saying (b, c) have been compacted into d, I can start work on compacting

a and d, even though the change to replace [a, b, c] with [a, d] hasn't been persisted yet. I think thats all fine, because, even if the commit to replace [a, b, c] with [a, d] fails, if it fails we just revert back to [a, b, c]. if we had send out a request to compact (a, d) before making that durable, we'll just get a response that we can't use which is also fine.

Reply:
right now, the speculative thing I described won't happen because we only look for potential requests in `step` after we've called drain_pending (so we can never issue two compaction requests without committing meta at least once in between). but you could imagine in the future we might issue compaction requests at a different cadence.